### PR TITLE
descrption: Functional component

### LINKED ISSUE #81: **Descriptions: Hide Validations**
- [x] Refactor **Comments** components to functional
- [x] Refactor **Comments** using state objects
- [x] Refactor **webapp/components/description/*** using state objects
- [x] **All tables** National Data and Analysis and processing of national data hide validation

### PR DESCRIPTION
**Pull request by sorja**
Resolves #81

### CODE REVIEW

File: webapp/components/description/dataSourceDescriptionAndComments.js [hunk 1]
```diff
@@ -34,8 +34,7 @@ const CommentableReviewDescriptions = props => {
           title={i18n.t('description.dataSourcesTitle')}
           section={section}
           name={dataSources}
-          template={dataSourcesEditorTemplate(i18n)}
-          countryIso={countryIso}/>
+          template={dataSourcesEditorTemplate(i18n)} />
```
Comment (minotogna) [inline webapp/components/description/dataSourceDescriptionAndComments.js]: - I would use useCountryIso hook
- I would use useSelector(ReviewState.getOpenThreadTarget)
- countryIso is not a required prop
- I would remove mapStateToProps

File: webapp/components/description/description.js [hunk 2]
```diff
@@ -1,148 +1,148 @@
...
 import Tooltip from '@webapp/components/tooltip'
 import ckEditorConfig from '@webapp/components/ckEditor/ckEditorConfig'
 
-import * as AppState from '@webapp/app/appState'
+import useI18n from '@webapp/components/hooks/useI18n'
+import useCountryIso from '@webapp/components/hooks/useCountryIso'
+import useUserInfo from '@webapp/components/hooks/useUserInfo'
+import * as DescriptionState from '@webapp/components/description/descriptionState'
 
 import { saveDescriptions, fetchDescriptions, openEditor, closeEditor } from './actions'
```
Comment (minotogna) [inline webapp/components/description/description.js]: use root imports

File: webapp/components/description/description.js [hunk 3]
```diff
@@ -1,148 +1,148 @@
...
+          section={section}
+          name={name}
+          template={template}
+          content={content}
+        />
+      )
+    if (content)
+      return <div className="fra-description__preview" dangerouslySetInnerHTML={{ __html: content }} />
+    if (template)
+      return <div className="fra-description__preview" dangerouslySetInnerHTML={{ __html: template }} />
```
Comment (minotogna) [inline webapp/components/description/description.js]: this is the same. can we have a variable ```text = content || template ```

File: webapp/components/description/description.js [hunk 4]
```diff
@@ -1,148 +1,148 @@
...
+  const userInfo = useUserInfo()
 
-  showEditorContent(isActive, showDash) {
+  useEffect(() => {
+    dispatch(fetchDescriptions(countryIso, section, name))
+  }, [])
+
+  useEffect(() => {
+    dispatch(fetchDescriptions(countryIso, section, name))
+  }, [countryIso])
```
Comment (minotogna) [inline webapp/components/description/description.js]: The first hook is not necessary.

File: webapp/components/description/description.js [hunk 5]
```diff
@@ -1,148 +1,148 @@
...
-import * as AppState from '@webapp/app/appState'
+import useI18n from '@webapp/components/hooks/useI18n'
+import useCountryIso from '@webapp/components/hooks/useCountryIso'
+import useUserInfo from '@webapp/components/hooks/useUserInfo'
+import * as DescriptionState from '@webapp/components/description/descriptionState'
 
 import { saveDescriptions, fetchDescriptions, openEditor, closeEditor } from './actions'
 
-class Description extends Component {
+const Description = props => {
```
Comment (minotogna) [inline webapp/components/description/description.js]: - I would extract the two components into separate files
- I would use propTypes and defaultProps

File: webapp/components/description/description.js [hunk 6]
```diff
@@ -1,148 +1,148 @@
...
+        showError ?
+          <Tooltip error text={i18n.t('generalValidation.emptyField')}>
+            {title} <Icon key="icon-error" className="icon-margin-left icon-red" name="alert" />
+          </Tooltip>
+          :
+          <>{title}</>
+      }
+    </h3>
+    {
+      !disabled && <div className="fra-description__link no-print"
```
Comment (minotogna) [inline webapp/components/description/description.js]: i would put the div to a new line

File: webapp/components/description/description.js [hunk 7]
```diff
@@ -1,148 +1,148 @@
...
+        onClick={e => {
+          isActive ? dispatch(closeEditor()) : dispatch(openEditor(name))
+          e.stopPropagation()
```
Comment (minotogna) [inline webapp/components/description/description.js]: why do u need e.stopPropagation after dispatching the action?

File: webapp/components/description/description.js [hunk 8]
```diff
@@ -1,148 +1,148 @@
...
+  </div>
+}
+
+Description.propTypes = {
+  section: PropTypes.string.isRequired
 }
 
-class DescriptionEditor extends Component {
+const DescriptionEditor = props => {
+  const {
```
Comment (minotogna) [inline webapp/components/description/description.js]: I would add propTypes

File: webapp/components/description/dataSourceDescriptionAndComments.js (hunk 1 above)
Comment (sorja) [reply webapp/components/description/dataSourceDescriptionAndComments.js]: Unused file

File: webapp/components/description/DescriptionEditor.js [hunk 9]
```diff
@@ -0,0 +1,48 @@
...
+  </div>;
+};
+
+DescriptionEditor.propTypes = {
+  name: PropTypes.string,
+  template: PropTypes.string,
+  section: PropTypes.string.isRequired,
+  content: PropTypes.string,
+}
+export default DescriptionEditor
```
Comment (minotogna) [inline webapp/components/description/DescriptionEditor.js]: we do not use ```;``` in the code

File: webapp/components/description/actions.js [hunk 10]
```diff
@@ -19,20 +17,18 @@ export const saveDescriptions = (countryIso, section, name, content) => dispatch
...
-  const dispatched = dispatch => {
-    return axios.post(`/api/country/descriptions/${countryIso}/${section}/${name}`, {content}).then(() => {
-      dispatch(autosave.complete)
-    }).catch((err) => {
-      dispatch(applicationError(err))
-    })
+  const dispatched = async dispatch => {
+    await axios.post(`/api/country/descriptions/${countryIso}/${section}/${name}`, { content })
+    dispatch(autosave.complete)
   }
```
Comment (minotogna) [inline webapp/components/description/actions.js]: shouldn't this file go under assessment/description ?

**Pull request merged**